packages/framework/presence/src/systemWorkspace.ts

```diff
@@ -52,67 +80,22 @@ class SystemWorkspaceImpl implements PresenceStatesInternal, SystemWorkspace {
 		for (const [clientConnectionId, value] of Object.entries(
 			remoteDatastore.clientToSessionId,
 		)) {
+			const clientSessionId = value.value;
+			const attendee = this.ensureAttendee(clientSessionId, clientConnectionId, value.rev);
 			const knownSessionId: InternalTypes.ValueRequiredState<ClientSessionId> | undefined =
 				this.datastore.clientToSessionId[clientConnectionId];
 			if (knownSessionId === undefined) {
 				this.datastore.clientToSessionId[clientConnectionId] = value;
 			} else {
 				assert(knownSessionId.value === value.value, "Mismatched SessionId");
 			}
-			value: this.selfAttendee.sessionId,
-		};
-
-		this.selfAttendee.currentConnectionId = () => clientConnectionId;
-		this.attendees.set(clientConnectionId, this.selfAttendee);
-	}
-
-	public getAttendees(): ReadonlySet<ISessionClient> {
-		return new Set(this.attendees.values());
-	}
-
-	public getAttendee(clientId: ClientConnectionId | ClientSessionId): ISessionClient {
-		const attendee = this.attendees.get(clientId);
-		if (attendee) {
-			return attendee;
-		}
-
-		throw new Error("Attendee not found");
-	}
-
-	public getMyself(): ISessionClient {
-		return this.selfAttendee;
-	}
-
-	private ensureAttendee(
-		clientSessionId: ClientSessionId,
-		clientConnectionId: ClientConnectionId,
-		order: number,
-	): SessionClient {
-		const currentConnectionId = (): ClientConnectionId => clientConnectionId;
-		let attendee = this.attendees.get(clientSessionId);
-		if (attendee === undefined) {
-			attendee = {
-				sessionId: clientSessionId,
-				order,
-				currentConnectionId,
-			};
-			this.attendees.set(clientSessionId, attendee);
-		} else if (order > attendee.order) {
-			attendee.order = order;
-			attendee.currentConnectionId = currentConnectionId;
+			if (!(clientSessionId in this.datastore.clientToSessionId)) {
```

Comment:
Im assuming this is the only place where local client learns about remote client's session ID? I have not looked under the hood, but Is this possibly lossy signal? If so, what happens on loss? What are the expectations between the time client (client id) shows up in the audience and its client session ID is unknown?

Reply:
Great call out. Before this change code was resilient to missing entry as any entry (which is added under a session id key) would create the session client on demand.
I think there is more metadata that I want to send with each update. We can always have "header" with each update that has the current connection id to session id entry for resiliency. (Overall resiliency patterns have not been implemented.)
Alternatively, we can restore the on-demand session client creation. I think I'll go with that for now as it covers other cases.

Reply:
On demand doesn't really work out. But sending single system workspace data for current connection is sufficient to ensure any update will be working with a known session client.